datafusion/core/tests/dataframe/mod.rs

```diff
@@ -1121,6 +1121,39 @@ async fn join() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn join_coercion_unnnamed() -> Result<()> {
+    let ctx = SessionContext::new();
+
+    // Test that join will coerce column types when necessary
+    // even when the relations don't have unique names
```

Comment:
Question: when the columns don't have unique names, is it safe to assume that the unqualified columns in `left_cols`/`right_cols` refer to the left/right relations, respectively? Since technically a valid join predicate is for example, `t1.id = t1.id`.

Seems like it would make sense to me most of the time! But there is some level of implicitness here.

Reply:
> when the columns don't have unique names, is it safe to assume that the unqualified columns in left_cols/right_cols refer to the left/right relations, respectively?

Yes, i think so: https://docs.rs/datafusion/latest/datafusion/dataframe/struct.DataFrame.html#method.join

Though perhaps we could make that explicit in the docs 🤔

Reply:
I think that would be a good idea! 🙌🏾

Reply:
Any chance you are willing to make a PR? I would be happy to review it :)

Reply:
Sure! Add a small doc update here https://github.com/apache/datafusion/pull/14393